tasks/vesting.ts

```diff
@@ -1,1028 +1,1069 @@
+  ): Promise<void> => {
+    const runSubtask = hre.run as RunVestingWithSubTasks;
+    const blockchainGrants = await runSubtask('get-blockchain', {
+      grants: { githubGrants },
+      lNori,
+    });
+    const grantDiffs: any[] = Object.values(
+      Object.fromEntries(
+        Object.entries(
+          diff(blockchainGrants, githubGrants, { full: true })
+        ).filter(([dk, d]: [any, any]) => {
+          return Object.entries(d).find(([k, v]: [any, any]) => {
+            const isDifferent =
+              !Boolean((blockchainGrants[dk] as any)?.exists) &&
+              Boolean(v) &&
+              k !== 'lastRevocationTime' &&
+              k !== 'lastQuantityRevoked' &&
+              v?.__new;
+            return isDifferent;
+          });
+        })
+      )
+    );
+    hre.log(
+      chalk.bold.bgWhiteBright.black(
+        `Found ${grantDiffs.length} grants that need updating`
+      )
+    );
+    const buildUserData = ({ grant }: { grant: any }): string => {
+      return hre.ethers.utils.defaultAbiCoder.encode(
+        [
+          'address',
+          'uint256',
+          'uint256',
+          'uint256',
+          'uint256',
+          'uint256',
+          'uint256',
+          'uint256',
+          'uint256',
+          'uint256',
+        ],
+        [
+          grant.recipient.__new ?? grant.recipient,
+          grant.startTime.__new ?? grant.startTime,
+          grant.vestEndTime.__new ?? grant.vestEndTime,
+          grant.unlockEndTime.__new ?? grant.unlockEndTime,
+          grant.cliff1Time.__new ?? grant.cliff1Time,
+          grant.cliff2Time.__new ?? grant.cliff2Time,
+          BigNumber.from(
+            grant.vestCliff1Amount.__new ?? grant.vestCliff1Amount
+          ),
+          BigNumber.from(
+            grant.vestCliff2Amount.__new ?? grant.vestCliff2Amount
+          ),
+          BigNumber.from(
+            grant.unlockCliff1Amount.__new ?? grant.unlockCliff1Amount
+          ),
+          BigNumber.from(
+            grant.unlockCliff2Amount.__new ?? grant.unlockCliff2Amount
+          ),
+        ]
+      );
+    };
+    if (grantDiffs.length > 0) {
+      let totalAmount = BigNumber.from(0);
+      const recipients = grantDiffs.map((_) => lNori.address);
+      const amounts = grantDiffs.map((grant) => {
+        const grantAmount = BigNumber.from(
+          grant.originalAmount.__new ?? grant.originalAmount
+        );
+        totalAmount = totalAmount.add(grantAmount);
+        return BigNumber.from(
+          grant.originalAmount.__new ?? grant.originalAmount
+        );
+      });
+      const userData = grantDiffs.map((grant) => buildUserData({ grant }));
+      const operatorData = grantDiffs.map((_) => '0x');
+      const requireReceptionAck = grantDiffs.map((_) => true);
+      hre.log(
+        `Total bpNORI to lock: ${ethers.utils.formatEther(
+          amounts.reduce((accumulator, v) => accumulator.add(v))
+        )}`
+      );
```

Comment:
hard to see the actual diff because of all the uncommenting in this file but below this comment is where the actual changes that involve the permit are (through line 906 where `batchCreateGrants` is also called in the dryrun situation)

Reply:
Nice!  Might be nice to pull the signing logic out into a separate function but other than that looks great!  I'll pull it this evening and try to sort out the named account thing.

Reply:
[there's a helper util for this in utils/math.ts](https://github.com/nori-dot-eco/contracts/blob/master/utils/math.ts#L7)